community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/executionplan/StandardInternalExecutionResult.scala

```diff
@@ -98,9 +99,13 @@ abstract class StandardInternalExecutionResult(context: QueryContext, runtime: R
   override def accept[E <: Exception](visitor: ResultVisitor[E]): Unit = {
     accept(new QueryResultVisitor[E] {
       override def visit(record: QueryResult.Record): Boolean = {
-        val row = new MapBasedRow
-        row.map = fieldNames().zip(record.fields().map(context.asObject)).toMap
-        visitor.visit(row)
+        val names = fieldNames()
```

Comment:
Probably not super important but `val names = fieldNames() ` can be done outside `visit`

Reply:
And I wonder if the fact that fieldNames is now a `val` would get this optimised by the compiler the same all all cases?